stage1/init/common/pod.go

```diff
@@ -256,24 +259,44 @@ func appToSystemd(p *stage1commontypes.Pod, ra *schema.RuntimeApp, interactive b
 		return errwrap.Wrap(errors.New("unable to write environment file"), err)
 	}
 
-	// This is a partial implementation for app.User and app.Group:
-	// For now, only numeric ids (and the string "root") are supported.
 	var uid, gid int
 	var err error
-	if app.User == "root" {
-		uid = 0
+
+	if strings.HasPrefix(app.User, "/") {
+		var stat syscall.Stat_t
+		if err = syscall.Lstat(strings.Join([]string{common.AppRootfsPath(p.Root, appName),
+			app.User}, "/"), &stat); err != nil {
+			return errwrap.Wrap(fmt.Errorf("unable to get uid from file %q",
+				app.User), err)
+		}
+		uid = int(stat.Uid)
```

Comment:
If rkt is called with `--private-users` (user namespace), I think you need to do some uid unshifting after the Lstat because the files from the ACI are extracted with a uid shift but rkt is still on the host user namespace while it is running `appToSystemd`.

Something like the following (untested):
```
uidRange := uid.NewBlankUidRange()
if err := uidRange.Deserialize([]byte(privateUsers)); err != nil {
    return err
}
if uid < uidRange.Shift || uid > uidRange.Shift + uidRange.Count {
    return err
}
uid = int(stat.Uid) - uidRange.Shift
```

But the code should be in the [uid package](https://github.com/coreos/rkt/blob/master/pkg/uid/uid.go#L62). It already has a function `ShiftRange()` but it does not have `UnshiftRange()` yet. It could be added.